docs/proposal/architecture_for_diff_iaas.md

```diff
@@ -0,0 +1,32 @@
+# IBM Cloud Provider with multiple types of IaaS
+
+## Motivation
+
+There are multiple types of IaaS in IBM Cloud: Classic infrastructure, VPC and PowerVS etc. Each type of IaaS has some specific features. Users would like to bootstrap a k8s cluster from any types of IaaS. It's hard to maintain an individual repository for each type of IaaS. A good implementation should describe IBM Cloud feature in an appropriate grain. 
+
+## Proposal
+
+This proposal presents a 2 layers infrastructure resource definition. Th top layer describes IBM Cloud common feature and the 2nd layer describes the details of the specific IaaS implementation.
+
+![](../images/cluster-api-ibmcloud-definition.png)
+
+IBMCluster/IBMMachine/IBMMachineTemplate is the generic top layer. It would only show the required information to reconsile status with cluster-api level Objects. Cluster, Machine and MachineDeployment etc. It doesn't know that VPC infrastructure would care about subnet details, public gateways, security groups. And it also doesn't care about PowerVS needs CloudInstance.
+
+We borrowed the `infrastructureRef` concept from cluster-api level Object. An IBMCluster should reference a specific IaaS implementation. Using VPC as an example we have an individual `IBMVPCCluster` object which has all the details that are needed in IBM Cloud VPC. There is also a specific ibmvpccluster_controller to watch the `IBMVPCCluster` and manuplate the VPC, subnet, security group and public gateway. They are all VPC resouces.
```

Comment:
I understand `infrastructureRef` will be used , it should be good to go this way


in the picture below ,who is responsible for define `IBMPowerVSCluster` and `IBMPowerVSMachine`?

Reply:
Should be PowerVS team